ngraph/core/src/op/loop.cpp

```diff
@@ -167,16 +168,26 @@ void op::v5::Loop::validate_and_infer_types()
             m_num_iterations = val[0];
     }
 
-    NODE_VALIDATION_CHECK(this,
-                          get_input_size() == m_input_descriptions.size() + 2,
-                          "Number of inputs must be the same as number of input descriptions");
+    // WA: input description with index 0 or 1 means that Loop consructor will duplicate it in
+    // the inputs.
+    // When using visit_attributes() no duplication occurs, input_offset shall be decremented.
+    size_t input_offset = 2;
+    for (const auto& in_desc : m_input_descriptions)
+    {
+        if (in_desc->m_input_index == 0 || in_desc->m_input_index == 1)
+        {
+            input_offset--;
+        }
+    }
+    // throw exception when input_offset < 0.
```

Comment:
too obvious comment :) 
should be something like:
 input_offset < 0 means that there are several duplications of external_port_id (the same ext_port_id is connected to several Parameters in the port map) in input_desc, this can lead to wrong or undefined behavior, so throw exception here. Ticket: ticket_number